std/math/algebraic.d

```diff
@@ -349,6 +353,13 @@ if (isFloatingPoint!T)
     else
         assert(0, "hypot not implemented");
 
+    if (u * T.epsilon > v)
+    {
+        // hypot (huge, tiny) = huge
+        // also: hypot(x, 0) = x
```

Comment:
And `hypot(0, x)`, as the first branch swaps the arguments so the left has the larger value.

Not sure if it's worth shortcutting `hypot(0,0)` too, as there's no problem with `1.0 * sqrt(0.0)` other than CPU time.